src/Raven.Studio/typescript/viewmodels/database/query/spatialQueryMap.ts

```diff
@@ -0,0 +1,104 @@
+import viewModelBase = require("viewmodels/viewModelBase");
+import spatialMapLayerModel = require("models/database/query/spatialMapLayerModel");
+import document = require("models/database/documents/document");
+import documentMetadata = require("models/database/documents/documentMetadata");
+import {Layer, MarkerClusterGroup} from "leaflet";
+
+class spatialQueryMap extends viewModelBase {
+    
+    geoData = ko.observableArray<spatialMapLayerModel>([]);
+
+    constructor(geoData: Array<spatialMapLayerModel>) {
+        super();
+        this.geoData(geoData);
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+        this.createMap();
+    }
+    
+    createMap() {
+
+        ////////////////////////////
+        // Define base tile layers
+        ////////////////////////////
+
+        const osmLink = `<a href="http://openstreetmap.org">OpenStreetMap</a>`;
+        const osmUrl = 'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png';
+        const osmAttrib = `&copy; ${osmLink} Contributors`;
+        const osmMap = L.tileLayer(osmUrl, {attribution: osmAttrib});
+
+        const otmLink = `<a href="http://opentopomap.org/">OpenTopoMap</a>`;
+        const otmUrl = `http://{s}.tile.opentopomap.org/{z}/{x}/{y}.png`;
+        const otmAttrib = `&copy; ${otmLink} Contributors`;
+        const otmMap = L.tileLayer(otmUrl, {attribution: otmAttrib});
+
+        const baseLayers = {
+            "Streets Map": osmMap,
+            "Topography Map": otmMap
+        };
+
+        //////////////////////////////////////////////////////////////
+        // Define group layers (layer per spatial point from query)
+        //////////////////////////////////////////////////////////////
+
+        let dataLayers = {};
+        let markersGroups: MarkerClusterGroup[] = [];
+
+        this.geoData().forEach((layerModel) => {
+            
+            let markers: Layer[] = [];
+            let markersGroup = L.markerClusterGroup();
+
+            const generatePopupContent = (doc: document) => {
+                const docDto = doc.toDto(true);
+                const metaDto = docDto["@metadata"];
+                documentMetadata.filterMetadata(metaDto);
+
+                let text = JSON.stringify(docDto, null, 4);
+                text = Prism.highlight(text, (Prism.languages as any)["javascript"]);
+
+                const textHtml = `<div>
+                                    <h4>Document: ${doc.getId()}</h4>
```

Comment:
escaping! Please remeber about that every time you concat user defined input with html tags.

Reply:
done